src/js/gutenberg-plugins/inserter-media-category.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Some code here was copied from Jetpack's implementation of the inserter media category.
+ * See https://github.com/Automattic/jetpack/pull/31914
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { dispatch, select, subscribe } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+const { classifaiDalleData } = window;
+
+const isInserterOpened = () =>
+	select( 'core/edit-post' )?.isInserterOpened() ||
+	select( 'core/edit-site' )?.isInserterOpened() ||
+	select( 'core/edit-widgets' )?.isInserterOpened?.();
+
+const waitFor = async ( selector ) =>
+	new Promise( ( resolve ) => {
+		const unsubscribe = subscribe( () => {
+			if ( selector() ) {
+				unsubscribe();
+				resolve();
+			}
+		} );
+	} );
+
+waitFor( isInserterOpened ).then( () =>
+	dispatch( 'core/block-editor' )?.registerInserterMediaCategory?.(
+		registerGenerateImageMediaCategory()
+	)
+);
+
+const registerGenerateImageMediaCategory = () => ( {
+	name: 'classifai-generate-image',
+	labels: {
+		name: classifaiDalleData.tabText,
+		search_items: __( 'Enter a prompt', 'classifai' ),
+	},
+	mediaType: 'image',
+	fetch: async ( { search = '' } ) => {
```

Comment:
I think this needs to be debounced a little, it looks like it makes multiple requests while typing in the prompt.

<img width="1122" alt="Screen Shot 2023-08-02 at 9 34 16 am" src="https://github.com/10up/classifai/assets/519727/5b986e90-6e89-47a0-9727-0d1026a71ecf">

Reply:
I mentioned that in the PR description that if someone doesn't type super fast, it will make multiple requests. I'm struggling with figuring out how to add debouncing here, since debouncing already happens on the WordPress side (see https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inserter/hooks/use-debounced-input.js). Everything I've tried to add another layer of debouncing isn't working. Either I'm not sure how to make this work (very likely) or this may not be possible since the only thing we have control over is the actual `fetch` function, not when that function gets called.

Reply:
Sorry Darin, I missed this in the description. 

As the feature didn't make it in to WP 6.3, is it worth adding a note that it requires the Gutenberg plugin/WP 6.4 or later to the changelog entry?

Reply:
I'm surprised this didn't make it into WP 6.3 as this changed was added to Gutenberg in min-June. I think this is probably worth holding off on until this comes to WordPress itself, at least that's my opinion.

I also think it's worth discussing a bit more around how requests are made here. Since image generation requests are expensive, it's not great that you'll almost certainly end up making multiple requests as you enter an image prompt. I'm wondering if it's worth opening an issue on the Gutenberg side to see if we can get this feature modified a bit? I would love to be able to toggle between search happening as you type (the way it works now) or the ability to only search if someone clicks a submit button (which would solve our needs). Or if we can filter the default debounce value to make that higher, that also would help.

Not sure if either of those are changes they'll deem worth adding but may be a good conversation to start.